### PR TITLE
gh-96819: check if the length of a pipe write is not greater than 512

### PR DESCRIPTION
**Pull request by saito828koki**
* Issue: gh-96819

### CODE REVIEW

File: Lib/multiprocessing/resource_tracker.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def unregister(self, name, rtype):
     def _send(self, cmd, name, rtype):
         self.ensure_running()
         msg = '{0}:{1}:{2}\n'.format(cmd, name, rtype).encode('ascii')
-        if len(name) > 512:
+        if len(msg) > 512:
             # posix guarantees that writes to a pipe of less than PIPE_BUF
             # bytes are atomic, and that PIPE_BUF >= 512
             raise ValueError('name too long')
```
Comment (DanielNoord) [inline Lib/multiprocessing/resource_tracker.py]: Shouldn't this also be updated? ie. `msg too long`?

Comment (saito828koki) [reply Lib/multiprocessing/resource_tracker.py]: @DanielNoord 
Thanks for your comment!
The error message should also be updated. I will fix it.

File: Misc/NEWS.d/next/Library/2022-09-17-13-15-10.gh-issue-96819.6RfqM7.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Fixed the block to guarantee that the length of a write to a pipe is not greater than PIPE_BUF
```
Comment (DanielNoord) [inline Misc/NEWS.d/next/Library/2022-09-17-13-15-10.gh-issue-96819.6RfqM7.rst]:
```suggestion
Fixed the check that guarantees that the length of a write to a pipe is not greater than PIPE_BUF
```

**Review by DanielNoord — APPROVED** [review]

File: Misc/NEWS.d/next/Library/2022-09-17-13-15-10.gh-issue-96819.6RfqM7.rst (hunk 2 above)
Comment (saito828koki) [reply Misc/NEWS.d/next/Library/2022-09-17-13-15-10.gh-issue-96819.6RfqM7.rst]: Thank you for correcting my English! Your suggestion seems to be great.

Comment (DanielNoord) [reply Misc/NEWS.d/next/Library/2022-09-17-13-15-10.gh-issue-96819.6RfqM7.rst]: No worries! I struggle with writing good and short changelog entries myself many times! 😄

**Comment by JelleZijlstra** [comment]
Is it possible to write a unit test for this change?

**Comment by saito828koki** [comment]
Hello @JelleZijlstra ,

Thanks for your comment.
I added a test for this change. Could you review this PR?

File: Misc/NEWS.d/next/Library/2022-09-17-13-15-10.gh-issue-96819.6RfqM7.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Fixed the check that guarantees that the length of a write to a pipe is not greater than PIPE_BUF
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Library/2022-09-17-13-15-10.gh-issue-96819.6RfqM7.rst]: Mention `multiprocessing.resource_tracker` in news.

**Review by kumaraditya303 — CHANGES_REQUESTED** [review]

File: Misc/NEWS.d/next/Library/2022-09-17-13-15-10.gh-issue-96819.6RfqM7.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Fixed the check in :mod:`multiprocessing.resource_tracker` that guarantees that the length of a write to a pipe is not greater than PIPE_BUF
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Library/2022-09-17-13-15-10.gh-issue-96819.6RfqM7.rst]:
```suggestion
Fixed check in :mod:`multiprocessing.resource_tracker` that guarantees that the length of a write to a pipe is not greater than ``PIPE_BUF``.
```

**Review by kumaraditya303 — APPROVED** [review]

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**